extensions/test/BUILD.gn

```diff
@@ -106,6 +106,16 @@ loadable_module("echo_extension") {
     "echo_extension.c",
   ]
   output_dir = "$root_out_dir/tests/extension/echo_extension"
+
+  if (is_android) {
```

Comment:
This is overly complicated. GN doesn't seem to complain about entries in `visibility` that do not exist, so you can do something like this instead:

```
loadable_module("echo_extension") {
  visibility = [
    ":*",

    # These test Android targets also package this library.
    "//xwalk/app/android/runtime_client_embedded_shell:*",
    "//xwalk/app/android/runtime_client_shell:*",
  ]
  sources = [
    "echo_extension.c",
  ]

  # This library is used as in |native_libs| in the Android targets mentioned
  # above, and as of M52 these libraries are required to be in $root_out_dir.
  # TODO(heke123): Check if we can get rid of this once we move to M53.
  if (!is_android) {
    output_dir = "$root_out_dir/tests/extension/echo_extension"
  }
}
```